src/test/java/server/actions/EtagGeneratorExceptionTest.java

```diff
@@ -0,0 +1,31 @@
+package server.actions;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.security.NoSuchAlgorithmException;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class EtagGeneratorExceptionTest {
+    private NoSuchAlgorithmException originalException;
+    private EtagGeneratorException exception;
+
+    @Before
+    public void setUp() throws Exception {
+        originalException = new NoSuchAlgorithmException("no sha 1 algorithm");
+        exception = new EtagGeneratorException("Error in Etag Generation", originalException);
```

Comment:
as the error message always seems to be "error in Etag generation", could that not have been embedded in the exception class, rather than passing it in?